implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -266,11 +279,11 @@ private void connectAndSubscribe(UdpClient udpClient) {
         }));
     }
 
-    private void retryReplaceClient(Mono<? extends Connection> connection) {
-         connection
+    private void retryReplaceClient(Mono<? extends Connection> connectMono) {
+         connectMono
                  .retryWhen(Retry.backoff(Long.MAX_VALUE, Duration.ofSeconds(1)).maxBackoff(Duration.ofMinutes(1)))
-                 .subscribe(client -> {
-                     this.client.update(client);
+                 .subscribe(connection -> {
+                     this.statsdConnection.replace(connection);
```

Comment:
As mentioned in the description, we should no longer need to dispose of the previous connection when a new one is made because we would only make a new connection if the previous one were already closed (either through error or the registry being stopped) or it is the first connection made. Connections were leaking before due to `doOnDisconnected` being called twice for the same connection, which is no longer an issue since we check for it having been called on a connection previously.